src/main/code.ts

```diff
@@ -5,125 +5,422 @@
+				}
+				var textFrame = getFirstChildOfTypeText(columnChildren[j])
+				if (textFrame != null && textFrame.type == 'TEXT') {
+					textFrames.push(textFrame)
+				}
+			}
 		}
+	}
+
+	return maxRowLength
+}
+
+function updateColumnCount(selectedFrame: FrameNode, columnLength: number, columnLengthData: number) {
+	if ("children" in selectedFrame) {
 
-		if (i == 0) {
-			headerCells.push(newNode)
+		if (columnLength > columnLengthData) {
+			// delete rows
+			for (var i = columnLengthData; i < columnLength; i++) {
+				selectedFrame.children[columnLengthData].remove()
+			}
+
+		} else if (columnLength < columnLengthData) {
+			// add rows
+			for (var i = columnLength; i < columnLengthData; i++) {
+				var copy = selectedFrame.children[columnLength - 1].clone()
+				selectedFrame.appendChild(copy)
+			}
 		}
-	  }
 	}
-	original.remove();
-	return headerCells;
-  }
-  
-  function getFirstChildOfTypeText(sceneNode: SceneNode) {
-	if ("children" in sceneNode) {
-	  return sceneNode.findOne(node => node.type === "TEXT") as TextNode
+}
+
+function updateRowCount(selectedFrame: FrameNode, columnLength: number, rowLength: number, rowLengthData: number) {
+	if ("children" in selectedFrame) {
+		if (rowLength > rowLengthData) {
+			// delete columns
+			for (var i = 0; i < columnLength; i++) {
+				var column = selectedFrame.children[i] as FrameNode
+				for (var j = rowLengthData; j < rowLength; j++) {
+					column.children[rowLengthData].remove()
+				}
+			}
+
+		} else if (rowLength < rowLengthData) {
+			// add columns
+			for (var i = 0; i < columnLength; i++) {
+				var column = selectedFrame.children[i] as FrameNode
+				for (var j = rowLength; j < rowLengthData; j++) {
+					var copy = column.children[rowLength - 1].clone()
+					column.appendChild(copy)
+				}
+			}
+
+		}
+	}
+}
+
+function insertData(textNodes: TextNode[], cellsData: String[], right_align_numbers: boolean) {
+	// Copy text from data cell into table
+	var arrayLength = Math.min(textNodes.length, cellsData.length)
+	for (var i = 0; i < arrayLength; i++) {
+		var textNode = textNodes[i]
+		textNode.characters = cellsData[i].trim()
+		updateAlignment(textNode, right_align_numbers)
+	}
+
+}
+
+function updateAlignment(textNode: TextNode, right_align_numbers: boolean) {
+	var FrameNode = textNode.parent as FrameNode
+	// Align as asked
+	if (right_align_numbers && !isNaN(Number(textNode.characters.replaceAll("'", "").trim()))) {
+		textNode.textAlignHorizontal = 'RIGHT'
+		if (FrameNode.layoutMode == 'HORIZONTAL') {
+			FrameNode.primaryAxisAlignItems = 'MAX'
+		} else if (FrameNode.layoutMode == 'VERTICAL') {
+			FrameNode.counterAxisAlignItems = 'MAX'
+		}
 	} else {
-	  return null
+		textNode.textAlignHorizontal = 'LEFT'
+		if (FrameNode.layoutMode == 'HORIZONTAL') {
+			FrameNode.primaryAxisAlignItems = 'MIN'
+		} else if (FrameNode.layoutMode == 'VERTICAL') {
+			FrameNode.counterAxisAlignItems = 'MIN'
+
```

Comment:
```suggestion
```

Reply:
We need a new alignment of text since the content of the cells changes. Why do want to delete this part? (I will change it soon, so the alignment only changes if the user checks the checkbox for right alignment)